src/vector.c

```diff
@@ -32,6 +32,9 @@ GIT_INLINE(int) resize_vector(git_vector *v, size_t new_size)
 {
 	void *new_contents;
 
+	if (new_size == 0)
+		return 0;
```

Comment:
One might tend to think that resizing to `0` might mean that we should deallocate the vector. But `resize_vector` is only called from `git_vector_init`, `git_vector_insert`, `git_vector_insert_sorted`, `git_vector_insert_null`, `git_vector_size_hint` and `git_vector_resize_to`. Only for the last two it _might_ make sense to do the deallocation dance, though.

- `git_vector_resize_to` is only used in `git_vector_set`, so the deallocation path could never be hit here.
- `git_vector_size_hint` will only do any work if the new size hint is bigger than `alloc_size`. So it also cannot ever hit the deallocation path.

So the dealloc would in fact be useless.

I think the only code path that can actually invoke `resize_vector(v, 0)` is `git_vector_size_hint`. And I can imagine a caller trivially invoking it e.g. if he wants to allocate a vector for a set of `n` items, where `n` might also be `0`.

Anyway, long story short: makes sense to me.